pkgs/development/libraries/zlib/default.nix

```
@@ -43,6 +43,10 @@ stdenv.mkDerivation (finalAttrs: {
       hash = "sha256-mpOyt9/ax3zrpaVYpYDnRmfdb+3kWFuR7vtg8Dty3yM=";
     };
 
+  patches = [
+    ./mingw-shared.patch
+  ];
+
```

fwiw you can probably reduce the # of rebuilds if you gate the patch.
```suggestion
  patches = lib.optional stdenv.hostPlatform.isMingw ./mingw-shared.patch;
```

---

Unsure if it really matters.

---

That has been suggested before on other cross build package fixes and was found to still cause rebuilds. I believe this is because previously it was null and the conditional will return an empty array when false. If there was already a list of patches, gating it like that would prevent rebuilds but when adding a new `patches` field it doesn't seem to have any beneficial effect, sadly.

---

If that's the case then you could try it as an optionalAttrs instead, or an `if ... then ... else null`

---

That is consistent with what I have been told in the past, but I haven't kept pace with the latest in the guidelines for that. Seeing as this commit has been languishing since December when it was first submitted and it keeps getting the the point of consensus and then ignored and not merged, I will do absolutely anything a reviewer says which might finally convince one of them to merge it at this point, though.

---

I would merge it if I could